spring-cloud-task-core/src/test/java/org/springframework/cloud/task/micrometer/TaskMetricsTests.java

```diff
@@ -43,15 +45,15 @@
 
 	@BeforeEach
 	public void before() {
-		Metrics.globalRegistry.getMeters().forEach(Metrics.globalRegistry::remove);
-		simpleMeterRegistry = new SimpleMeterRegistry();
-		Metrics.addRegistry(simpleMeterRegistry);
-		taskMetrics = new TaskMetrics();
+		this.simpleMeterRegistry = (SimpleMeterRegistry) new SimpleMeterRegistry().withTimerObservationHandler();
+		this.taskMetrics = new TaskMetrics(this.simpleMeterRegistry);
 	}
 
 	@AfterEach
 	public void after() {
-		Metrics.globalRegistry.getMeters().forEach(Metrics.globalRegistry::remove);
+		this.simpleMeterRegistry.clear();
+		this.simpleMeterRegistry.setCurrentObservation(null);
+		ObservationRegistryAssert.assertThat(this.simpleMeterRegistry).doesNotHaveRemainingObservation();
```

Comment:
Actually since you've set this to null one line above this will be always true. You should remove line no 55 and leave this as a check that no observations got leaked